traffic_monitor/tests/gocover.bash

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#----------------------------------------
+set -e
+go get "github.com/wadey/gocovmerge"
+touch result.txt
+packages=( "$@" )
+coverage_out=()
+i=1
+for pkg in ${packages[@]} ; do
+    for d in $(go list $pkg | grep -v vendor); do
+        file="$i.out"
+        go test -v -coverprofile=$file $d | tee -a result.txt
+        cat result.txt
+        if [ -f $file ]; then
+            coverage_out+=( $file )
+        fi
+        ((i++))
```

Comment:
nope it didn't work

Reply:
Yes, besides that fact that `${@[@]}` should just be `$@` you'd still need a temporary file to avoid clobbering yourself. Could do
```bash
tmp="$(mktemp)"
gocovmerge coverage_out "$file" > "$tmp"
mv -f "$tmp" ./coverage_out
```
Which isn't a *whole* lot better - although it is just one extra file per package instead of N for N packages -, so if you don't think it's worth rewriting for that I'm good leaving it as-is.

This'd be a lot easier if `gocovmerge` could take input on stdin.

Reply:
gocovermerge doesn't do merge incrementally or from results appended in one file. I tried them first and then got down to what is there now.

Reply:
Yes it does:
```bash
#!/usr/bin/bash
# $GOPATH/src/github.com/apache/trafficcontrol/traffic_monitor/gotestcov.sh
coverage_out=()
i=1

for d in $(go list ./... ../lib/... | grep -v vendor); do
	file="$i.out"
	go test -v -coverprofile="$file" "$d"
	if [ -f "$file" ]; then
		coverage_out+=($file)
	fi
	((i++))
done

gocovmerge ${coverage_out[*]} > coverage.out

touch coverage_out
for pkg in $(go list ./... ../lib/... | grep -v ); do
	tmp="$(mktemp)"
	go test -v -coverprofile covProfile "$pkg" > "$tmp"
	mv -f "$tmp" coverage_out
done
```
then:
```shell
$ chmod a+x ./gotestcov.sh
$ ./gotestcov.sh > /dev/null
$ diff coverage_out coverage.out
$ wc -l coverage.out
3454 coverage.out
```